src/MobileDelivery/MobileController.js

```diff
@@ -4,184 +4,196 @@ import MobileStart from './MobileStart';
+            let dgObj = await db.ref(`accounts/${deliveryData.delivererGroup}`).once('value');
+            deliveryData.delivererGroup = dgObj.val().name;           
+        } else { // Emergency Pick Up: deliveryData.delivererGroup == null
+            deliveryData.delivererGroup = '';
+            deliveryData.deliverers = [];
+        }
+
+        this.setState({ deliveryObj: this.getDeliveryObj(deliveryData)});
+    }
+    
+    getDeliveryObj(rawDelivery) {
+        let deliveryObj = ( // pick entries from rawDelivery
+            ({ description, delivererGroup, deliverers, notes, isEmergency, status, pickedUpInfo, deliveredInfo }) => 
+                ({ description, delivererGroup, deliverers, notes, isEmergency, status, pickedUpInfo, deliveredInfo })
+        )(rawDelivery);
+
+        let startTimeObj = moment.unix(rawDelivery.startTimestamp);
+        let endTimeObj = moment.unix(rawDelivery.endTimestamp);
+
+        deliveryObj.date = startTimeObj.format('YYYY-MM-DD');
+        deliveryObj.startTime = startTimeObj.format('HH:mm');
+        deliveryObj.endTime = endTimeObj.format('HH:mm');
+        return deliveryObj;
     }
 
-    // TODO: Save completed delivery
-    saveValues(fields) {
-        this.setState({
-            currentDelivery: Object.assign({}, this.state.currentDelivery, fields)
+    async fetchDa(rawDa) {
+        let daPromise = new Promise( async (resolve, reject) => {
+            let daSnapshot = await db.ref(`donating_agencies/${rawDa.agency}`).once('value');
+            resolve(daSnapshot.val());
+        });
+
+        let daMemberPromise = new Promise( async (resolve, reject) => {
+            let daMemberSnapshot = await db.ref(`accounts/${rawDa.primaryContact}`).once('value');
+            resolve(daMemberSnapshot.val());
         });
+
+        let daData = await Promise.all([daPromise, daMemberPromise]);
+        this.aggrDa(daData[0], daData[1]);
     }
 
-    showStep() {
-        switch (this.state.step) {
-        default:
+    aggrDa(daData, daMemberData) {
+        let pickedPrimaryContact = (
+            ({ position, name, email, phone }) => ({ position, name, email, phone })
+        )(daMemberData);
+        
+        let daObj = {
+            primaryContact: pickedPrimaryContact,
+            agency: daData.name,
+            address: this.aggrAddress(daData.address),
+        };
+        this.setState({ donatingAgency: daObj});
+    }
+
+    async fetchRa(rawRa) {
+        let raSnapshot = await db.ref(`accounts/${rawRa.agency}`).once('value');
+        this.aggrRa(raSnapshot.val(), rawRa.primaryContact);
+    }
+
+    aggrRa(raMeta, raPrimaryContact) {
+        let raObj = {
+            primaryContact: raPrimaryContact,
+            agency: raMeta.name,
+            notes: raMeta.deliveryNotes,
+            address: this.aggrAddress(raMeta.address),
+        };
+        this.setState({ receivingAgency: raObj});
+    }
+
+    aggrAddress(rawAddress) {
```

Comment:
ah, i see. They aren't actually different, an `address` object is `{street1, street2, city, state, zipcode, officeNo}`, they might look like they are different because `street2` and `officeNo` are optional, so if they are null, they just don't exist in firebase